lib/cloud/aws/identity.go

```diff
@@ -52,6 +61,20 @@ type Role struct {
 	identityBase
 }
 
+// String returns the AWS identity ARN for the role. If the role is
+// an assumed-role, it is converted to the IAM role ARN instead of the
+// assumed-role ARN. This is necessary since identity.String() is used in
+// AWS IAM policy statements as the resource, and assumed-role is invalid
+// for those statements.
```

Comment:
since Steve pointed out above that it can't reconstruct IAM role ARNs from assumed-role ARN if the role has a path in it, I'm reverting this.

Assumed role can be the target only when the user is running as an assumed role and doesn't pass `--role` or `--user` to the configurator commands. In that case, I'll try to use IAM API to get their role ARN via `iam.GetRole(..RoleName: identity.GetName())`. If that fails, then the configurator doesn't have enough info to bootstrap their role's permissions, in which case the error message should tell the user to specify `--role` or `--user` with full path (if it has a path), or with full ARN.

Reply:
reverted this in 76da330ecf78feab5b09d0f2479ad9f5c7fb7dfc

and replaced with functionality that tries to get the role arn from AWS IAM here: 3eafd129a94d83092762ca82fd7add29a1ee4b6e